code/modules/mining/equipment_locker.dm

```diff
@@ -869,16 +879,17 @@
 		if(M.scan_state)
 			minerals += M
 	if(minerals.len)
-		for(var/mob/user in mobs)
-			if(user.client)
-				var/client/C = user.client
-				for(var/turf/simulated/mineral/M in minerals)
-					var/turf/F = get_turf(M)
-					var/image/I = image('icons/turf/smoothrocks.dmi', loc = F, icon_state = M.scan_state, layer = 18)
-					C.images += I
-					spawn(30)
-						if(C)
-							C.images -= I
+		for(var/turf/simulated/mineral/M in minerals)
+			var/obj/effect/mining_overlay/C = PoolOrNew(/obj/effect/mining_overlay, M)
+			C.icon_state = M.scan_state
+			spawn(30)
+				qdel(C)
+
+/obj/effect/mining_overlay
```

Comment:
use /obj/effect/overlay/temp instead because you're copying a lot of the stuff those do(also they return QDEL_HINTPUTINPOOL or whatever it's called and this doesn't so it won't pool right)